pallets/automation-time/src/tests.rs

```diff
@@ -267,6 +267,49 @@ fn schedule_xcmp_works() {
 	})
 }
 
+#[test]
+fn schedule_xcmp_through_proxy_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let provided_id = vec![50];
+		let alice = AccountId32::new(ALICE);
+		let bob = AccountId32::new(BOB);
+		let call: Vec<u8> = vec![2, 4, 5];
+
+		// Funds including XCM fees
+		get_xcmp_funds(bob.clone());
+
+		assert_ok!(AutomationTime::schedule_xcmp_task_through_proxy(
+			RuntimeOrigin::signed(bob.clone()),
+			provided_id,
+			ScheduleParam::Fixed { execution_times: vec![SCHEDULED_TIME] },
+			PARA_ID.try_into().unwrap(),
+			NATIVE,
+			MultiLocation::new(1, X1(Parachain(PARA_ID.into()))).into(),
+			call.clone(),
+			Weight::from_ref_time(100_000),
+			alice.clone(),
+		));
```

Comment:
There’s no connection between the wallets alice and bob. We are not checking if one is the other’s proxy wallet, correct?

Reply:
This is the implementation of `MockEnsureProxy`, which is only used for automation-time testing purposes.

https://github.com/OAK-Foundation/OAK-blockchain/blob/e8415385756db9daf29cf12cb2ae49e2772d4c7d/pallets/automation-time/src/mock.rs#L352

Reply:
Okay, so these two accounts are linked. This question is resolved.
```
if _delegator == ALICE.into() && _delegatee == BOB.into() {
			Ok(())
		}
```